onnxruntime/core/providers/cpu/tensor/cast_op.cc

```diff
@@ -308,8 +386,16 @@ Status Cast::Compute(OpKernelContext* context) const {
     return Status::OK();
   }
 
-  utils::MLTypeCallDispatcherFromTypeList<EnabledSrcTypes> dispatcher{from};
-  dispatcher.Invoke<SrcDispatcher>(to_, *context, shape, *X, *Y);
+  if (to_ != ONNX_NAMESPACE::TensorProto::FLOAT8E4M3FN &&
+      to_ != ONNX_NAMESPACE::TensorProto::FLOAT8E4M3FNUZ &&
+      to_ != ONNX_NAMESPACE::TensorProto::FLOAT8E5M2 &&
+      to_ != ONNX_NAMESPACE::TensorProto::FLOAT8E5M2FNUZ) {
+    utils::MLTypeCallDispatcherFromTypeList<EnabledSrcTypes> dispatcher{from};
+    dispatcher.Invoke<SrcDispatcherStd>(to_, *context, shape, *X, *Y);
```

Comment:
would it be simpler if we just add a saturate parameter to all dispatch targets and ignore where not applicable?

Reply:
I hesitated. I did not want to add an unused parameter in most cases.

Reply:
that's fair. I think having two versions of the dispatchers does further complicate an already complicated setup though.